R/chop.R

```diff
@@ -141,17 +141,17 @@ unchop <- function(data, cols, keep_empty = FALSE, ptype = NULL) {
 #   used to slice the data frame `x` was subset from to align it with `val`.
 # - `val` the unchopped data frame.
 
-df_unchop <- function(x, ..., ptype = NULL, keep_empty = FALSE) {
+df_unchop <- function(x, ..., ptype = NULL, keep_empty = FALSE, call = caller_env()) {
```

Comment:
Our emerging convention when passing error calls through as arguments seems to be:

- Name them `call` and `arg` when the function's main purpose is for some kind of input checking, i.e. `vec_assert()`, `check_*()` functions, `abort()`, etc.

- Name them `error_call` and `error_arg` otherwise.

Since `df_unchop()`'s main purpose is unchopping, it should use `error_call`. The rest of the PR probably needs a pass to do this updating too. I'll try and mark them as I see them.

We have started doing this in vctrs, tidyselect, internally in dplyr, and I've done it in ivs, so probably best to take the time to be consistent here since call improvements is what this PR is all about.

Reply:
This means we need to rename a lot of already exported functions? Isn't it a bit late to change this now?

By the way I don't consider a dotted variant to represent a different name. I see it as different syntax for the same name.

Reply:
Right, I don't think of them as different names either, but pragmatically you have to remember which of 4 options it is.